### PR TITLE
WIP [Tags] New Tag (& Label) Component

### PR DESCRIPTION
**Pull request by sampi**
@tynandebold @zdlm @sampi

Fixes issue #484 

Things missing/broken:

* [ ] Finalized design/simplification from @mihneaz 
* [x] UserImage vertical alignment
* [x] Delete button size and alignment
* [x] Clickable Tag with Delete button
* [x] Lock icon size and alignment
* [x] List of Tags using the JS API 
* [x] Gallery docs page
* [x] Add examples to the main components' page

### CODE REVIEW

**Comment by tynandebold** [comment]
@mihneaz I've finished each item here, except for the 'Finalized design/simplification' one. You can take a look at what's pushed to this branch and see if everything's good to go.

cc @zdlm @sampi

File: src/spiritual/spiritual-gui/gui-spirits@wunderbyte.com/spirits/gui.Spirit.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ gui.Spirit = gui.Class.create(
 		onconstruct: function() {},
 
 		/**
-		 * `onconfigure` gets callend immediately after construction. This
+		 * `onconfigure` gets callen immediately after construction. This
```
Comment (sampi) [inline src/spiritual/spiritual-gui/gui-spirits@wunderbyte.com/spirits/gui.Spirit.js]: typo fix has another typo =)

File: docs/dist/index.html [hunk 2]
```diff
@@ -11,7 +11,7 @@
     <script src="/dist/assets/jquery-2.2.4.min.js"></script>
     <link rel="stylesheet" href="//d5wfroyti11sa.cloudfront.net/prod/client/ts-10.0.28.min.css">
     <link rel="stylesheet" href="/dist/assets/dox.css">
-    <link rel="icon" href="data:;base64,iVBORw0KGgo=">
+    <link rel="icon" href="https://developers.tradeshift.com/icons/favicon-32x32.png">
```
Comment (sampi) [inline docs/dist/index.html]: We should probably move this icon to be part of our codebase, either base64 or just deploy the file.

Maybe we should have our own icon which is similar to it, but has some sort of `ts.ui` feel. What do you think @mihneaz ?

File: docs/index.html [hunk 3]
```diff
@@ -2,7 +2,7 @@
 <html class="ts-loading">
 	<head>
 		<title>Tradeshift UI</title>
-		<link rel="icon" href="data:;base64,iVBORw0KGgo="/>
+		<link rel="icon" href="https://developers.tradeshift.com/icons/favicon-32x32.png"/>
```
Comment (sampi) [inline docs/index.html]: We should probably move this icon to be part of our codebase, either base64 or just deploy the file.

Maybe we should have our own icon which is similar to it, but has some sort of ts.ui feel. What do you think @mihneaz ?

**Comment by sampi** [comment]
Looking good pending @mihneaz 's approval.

We can figure out deployment of `v10.1.0`, so don't merge it before we've bumped everything!

_Also, I can't approve the PR, because I opened it._

**Pull request merged**